ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/Util/AcidState.hs

```diff
@@ -101,3 +107,51 @@ zoomAll l upd = StateT $ \large -> do
 mustBeRight :: Either Void b -> b
 mustBeRight (Left  a) = absurd a
 mustBeRight (Right b) = b
+
+-- | Creates checkpoint of the current state of Acid DB,
+-- archive previous checkpoints and remove too old archives.
+-- This is needed to avoid storing all logs permanently.
+cleanupAcidState
+    :: forall m st. (MonadIO m, MonadMask m)
+    => AcidState st
+    -> FilePath
+    -> Int
+    -> (Severity -> Text -> m ())
+    -> m ()
+cleanupAcidState db path numberOfStoredArchives writeLog = perform
+    where
+      perform = cleanupAction `catchAny` handler
+      cleanupAction = do
+          writeLog Debug "Starting cleanup"
+          -- checkpoint/archive
+          liftIO $ createCheckpoint db >> createArchive db
+          writeLog Debug "Created checkpoint/archived"
+          -- cleanup old archive data
+          void $ flip catchAny (\e -> putStrLn @Text $ "Got error while cleaning up archive: " <> show e) $ do
+              removed <- liftIO $ cleanupOldAcidArchives path numberOfStoredArchives
+              writeLog Debug $ "Removed " <> pretty removed <> " old archive files"
+          pass
+
+      handler :: SomeException -> m ()
+      handler e = do
+          let report = do
+                  writeLog Error $ sformat ("acidCleanupWorker failed with error: "%shown%
+                                             " restarting in 1m") e
+                  threadDelay @Second 60
+          report `finally` perform
+
+-- Returns how many files were deleted
+cleanupOldAcidArchives
+    :: FilePath  -- ^ path with stored files
+    -> Int    -- how many files should be stored
+    -> IO Int -- how many files have been deleted
```

Comment:
Seems like this function is used only once, and hence it could go into the `where` block under `cleanupAcidState`.